transcom/mymove
# [MB-16340] Calculating the Required Delivery Date for the same ZIP 5s should enforce a 1 mile minimum

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16340?atlOrigin=eyJpIjoiYjk0MWIwNjU4MWY5NGIzZWJmYjBmNjJhY2IzZTdmNGIiLCJwIjoiaiJ9)

## Summary
Added a check in hhg_planner.go so if a move has same origin and destination zips, it would return a distance of 1. This fix resolves the issue of the Prime not being able to update pickup dates and shipment weights because the planner would return a distance of 0.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Create an HHG shipment as a customer with the same postal code value for the origin and destination addresses.
2. Approve move as Service Counselor.
3. Approve the move as the TOO
4. Update the Prime scheduled pickup dates and shipment weight information.
5. You should see the green success flag that states "Successfully updated shipment". 

### Backend
